lib/Backend/GlobOpt.cpp

```diff
@@ -4528,6 +4529,23 @@ GlobOpt::CollectMemOpInfo(IR::Instr *instr, Value *src1Val, Value *src2Val)
         // Fallthrough if not an induction variable
     }
     default:
+        // List of instruction that are valid with memop (ie: instr that gets removed if memop is emitted)
+        if (
+            this->currentBlock != loop->GetHeadBlock() &&
```

Comment:
Could we extract the opcode list as an API? That would probably make it friendlier for maintenance down the road. Otherwise LGTM.